### PR TITLE
Add SSL Option to Backend Service

### PR DESCRIPTION
**Pull request by KaiReichart**
This PR allows the backend service to be secured with SSL.  
It is completely optional, if the environment variables are not set the service will start up as normal without SSL.

To enable this, set the following environment variables:

```
BUSOLA_SSL_ENABLED=1
BUSOLA_SSL_KEY_FILE=/path/to/tls.key
BUSOLA_SSL_CRT_FILE=/path/to/tls.crt
```

Changes proposed in this pull request:

- Add the option to secure the backend service with SSL

### CODE REVIEW

**Comment by Wawrzyn321** [comment]
Hi @KaiReichart, thanks for your contribution! To get this thing running on CI, you need to also bump the k8s deployment here - https://github.com/kyma-project/busola/blob/main/resources/backend/deployment.yaml#L23. Just use your PR number.

**Comment by KaiReichart** [comment]
hey @Wawrzyn321, 
thanks for the pointer, have updated it but unfortunately can't trigger the retest

**Comment by mrCherry97** [comment]
Hi @KaiReichart 
we have a problem with linter check, please run `npm run lint-fix` in main catalog, there will be that changes:
image
image

After that, we can run tests on your PR.
I try to do it myself but I don't have access to your repo.
Thanks!

**Comment by KaiReichart** [comment]
hey @mrCherry97, just coming back from holiday! 
I added the changes you suggested, can you try the tests again pls?

**Comment by Wawrzyn321** [comment]
@KaiReichart Hello again. I regret to inform you that after a longer discussion with my product owner and team we won't be merging this - we don't need this SSL option for our Busola hosting, and (even if it's an opt-in thing) we'd need to provide full future support for this.

I hope running Busola from your fork will be fine and just enough for you. Thank you for your effort and good intentions.

**Comment by valentinvieriu** [comment]
@KaiReichart Thank you for reaching up on Slack Channel. 
Your use-case seems to be a valid one. Can you adjust the documentation and maybe provide a istio configuration that will work with https https://github.com/kyma-project/busola/tree/main/resources/istio

**Comment by KaiReichart** [comment]
@valentinvieriu @Wawrzyn321 @mrCherry97 
Thanks for your input, I added some more documentation and a sample `DestinationRule` for the istio configuration.  
Is this sufficient? Or should I go into more detail in the docs?

**Review by Wawrzyn321 — DISMISSED** [review]

File: docs/install-kyma-dashboard-manually.md [hunk 1]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
 
 3. Wait a minute for the Dashboard to start.
 4. The Dashboard is available under your domain (for example, `https://busola.local.kyma.dev/`).
+
+## Enable TLS in the backend service
+
+By default, the communication with the backend pod is plain HTTP. Even when you use istio service mesh, the last bit of communication
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:41]:
```suggestion
By default, the communication with the backend Pod is plain HTTP. Even when you use Istio Service Mesh, the last bit of communication between the Istio sidecar and the backend service stays plain HTTP.
```

Capitalization as in [Kyma style guide](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#capitalized-terms).
Let's not split a sentence in two lines.

File: docs/install-kyma-dashboard-manually.md [hunk 2]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+between the istio sidecar and the backend service stays plain HTTP.
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:42]:
```suggestion
```

File: docs/install-kyma-dashboard-manually.md [hunk 3]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+
+If your use-case requires TLS all the way into the backend pod, you can enable it by providing a TLS certificate to the backend service,
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:44]:
```suggestion
If your use case needs TLS all the way into the backend Pod, enable it by providing a TLS certificate to the backend service, setting some environment variables and changing the deployment.
```

File: docs/install-kyma-dashboard-manually.md [hunk 4]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+setting some environment variables and changing the deployment.
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:45]:
```suggestion
```

File: docs/install-kyma-dashboard-manually.md [hunk 5]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+
+1. Generate TLS certificate
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:47]:
```suggestion
1. Generate a TLS certificate.
```

Is any TLS certificate good, or does it have to fulfil any special requirements regarding encoding etc?

File: docs/install-kyma-dashboard-manually.md [hunk 6]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+2. Mount your certificate into the backend pod
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:48]:
```suggestion
2. Mount your certificate into the backend Pod.
```

File: docs/install-kyma-dashboard-manually.md [hunk 7]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+3. Set the following environment variables on the backend pod:
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:49]:
```suggestion
3. Set the following environment variables on the backend Pod:
```

File: docs/install-kyma-dashboard-manually.md [hunk 8]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+
+   ```
+   BUSOLA_SSL_ENABLED=1
+   BUSOLA_SSL_KEY_FILE=/path/to/tls.key
+   BUSOLA_SSL_CRT_FILE=/path/to/tls.crt
+   ```
+
+4. If you are using istio service mesh, enable the `destinationrule-busola-backend.yaml` resource in `resources/istio/kustomization.yaml`
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:57]:
```suggestion
4. If you are using Istio Service Mesh, enable the `destinationrule-busola-backend.yaml` resource in `resources/istio/kustomization.yaml`.
```

File: docs/install-kyma-dashboard-manually.md [hunk 9]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
+5. Install Busola according to [Install Kyma Dashboard with Istio Ingress](#install-kyma-dashboard-with-istio-ingress)
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:58]:
```suggestion
5. Install Busola according to [Install Kyma Dashboard with Istio Ingress](#install-kyma-dashboard-with-istio-ingress).
```

File: docs/install-kyma-dashboard-manually.md [hunk 10]
```diff
@@ -35,3 +35,24 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
...
 4. The Dashboard is available under your domain (for example, `https://busola.local.kyma.dev/`).
+
+## Enable TLS in the backend service
+
+By default, the communication with the backend pod is plain HTTP. Even when you use istio service mesh, the last bit of communication
+between the istio sidecar and the backend service stays plain HTTP.
+
+If your use-case requires TLS all the way into the backend pod, you can enable it by providing a TLS certificate to the backend service,
+setting some environment variables and changing the deployment.
+
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:46]: Are there any prerequisites to these instructions?

**Review by Wawrzyn321 — DISMISSED** [review]

File: docs/install-kyma-dashboard-manually.md [hunk 11]
```diff
@@ -38,21 +38,53 @@ To run Kyma Dashboard manually on a cluster, you must install it together with I
 
 ## Enable TLS in the backend service
 
-By default, the communication with the backend pod is plain HTTP. Even when you use istio service mesh, the last bit of communication
-between the istio sidecar and the backend service stays plain HTTP.
+By default, the communication with the backend Pod is plain HTTP. Even when you use Istio Service Mesh, the last bit of communication between the Istio sidecar and the backend service stays plain HTTP.
 
-If your use-case requires TLS all the way into the backend pod, you can enable it by providing a TLS certificate to the backend service,
-setting some environment variables and changing the deployment.
+If your use case needs TLS all the way into the backend Pod, enable it by providing a TLS certificate to the backend service, setting some environment variables and changing the deployment. Depending on your setup, you might have to use a valid TLS certificate (e.g. if you are directly exposing the backend Pod without TLS termination in between). In this case you might want to use cert-manager to handle your certificates or provide a valid one here and manage it yourself.
```
Comment (NHingerl) [inline docs/install-kyma-dashboard-manually.md:43]:
```suggestion
If your use case needs TLS all the way into the backend Pod, enable it by providing a TLS certificate to the backend service, setting some environment variables and changing the deployment. Depending on your setup, you might have to use a valid TLS certificate (for example, if you are directly exposing the backend Pod without TLS termination in between). In this case, you might want to use cert-manager to handle your certificates, or provide a valid one here and manage it yourself.
```

**Review by NHingerl — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**